### PR TITLE
[MNG-7112] --projects flag should honor --non-recursive flag

### PR DESCRIPTION
**Pull request by MartinKanters**
Following this checklist to help us incorporate your 
contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/MNG-7112) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without 
       pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[MNG-XXX] - Fixes bug in ApproximateQuantiles`,
       where you replace `MNG-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the 
       commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will 
       be performed on your pull request automatically.
 - [x] You have run the [Core IT][core-its] successfully.

If your pull request is about ~20 lines of code you don't need to sign an
[Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) if you are unsure
please ask on the developers list.

To make clear that you license your contribution under 
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [x] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

[core-its]: https://maven.apache.org/core-its/core-it-suite/

Related integration tests: https://github.com/apache/maven-integration-testing/pull/104

### CODE REVIEW

File: maven-core/src/main/java/org/apache/maven/graph/DefaultGraphBuilder.java [hunk 1]
```diff
@@ -284,20 +284,7 @@ public DefaultGraphBuilder( BuildResumptionDataRepository buildResumptionDataRep
             excludedProjects.addAll( getProjectsBySelectors( request, projects, optionalSelectors, false ) );
 
             result = new ArrayList<>( projects );
-
-            for ( MavenProject excludedProject : excludedProjects )
```
Comment (michael-o) [inline maven-core/src/main/java/org/apache/maven/graph/DefaultGraphBuilder.java:288]: Wasn't this removed in an another commit recently?

File: maven-core/src/test/java/org/apache/maven/graph/DefaultGraphBuilderTest.java [hunk 2]
```diff
@@ -390,6 +400,7 @@ private Dependency toDependency( MavenProject mavenProject )
         private String resumeFrom = "";
         private String makeBehavior = "";
         private File requestedPom = new File( PARENT_MODULE, "pom.xml" );
+        private boolean isRecursive = true;
```
Comment (michael-o) [inline maven-core/src/test/java/org/apache/maven/graph/DefaultGraphBuilderTest.java:403]: Shouldn't this just be `recursive`?

File: maven-core/src/main/java/org/apache/maven/graph/DefaultGraphBuilder.java (hunk 1 above)
Comment (mthmulders) [reply maven-core/src/main/java/org/apache/maven/graph/DefaultGraphBuilder.java:288]: I think you mean [this one](https://github.com/apache/maven/pull/458)? Indeed - we'll rebase one on top of the other, depending on the order in which we merge them.

Comment (michael-o) [reply maven-core/src/main/java/org/apache/maven/graph/DefaultGraphBuilder.java:288]: OK, makes sense.

File: maven-core/src/test/java/org/apache/maven/graph/DefaultGraphBuilderTest.java (hunk 2 above)
Comment (MartinKanters) [reply maven-core/src/test/java/org/apache/maven/graph/DefaultGraphBuilderTest.java]: Yep, I agree. It's changed now, and I've found and fixed some other inconsistencies around that parameter as well. Ready to be reviewed again.

**Review by michael-o — APPROVED** [review]

**Pull request closed without merging**